### ISSUE TITLE
onClick returns wrong lat and lng

### ISSUE THREAD

**Issue by johnhult** [google-map-react/google-map-react#614]
Hi,
I've been using this library and I've ran into some weird issues which I can't seem to find a solution on and even though the library works good it is lacking a bit in documentation and I can't seem to find any information about this particular problem.

First my code.

`addPoint = ({ x, y, lat, lng, event }) => {
		console.log(x, y, lat, lng, event);
		this.map.panTo({
			lat: lat,
			lng: lng
		});
		let mark = new this.maps.Marker({
			map: this.map,
			position: { lat: lat, lng: lng }
		});
}`

`<GoogleMapReact
						...
						onClick={this.addPoint}
                                                  ...
					>`

And this is what happens:

![map-issue-small](https://user-images.githubusercontent.com/1528648/42736195-2bb1e630-8862-11e8-8888-58baf2f91e23.gif)

So basically, the lat lng gets calculated from the corner on "onClick". So when I press the upper left corner, it puts the marker in the center. If I press the center, it puts the marker in the lower right corner.

Is it some sort of setting that has to be activated or what is the reason for this? It seems like a bug. Obviously I want the panTo function to center on where it's clicked and of course I want the marker to be where I actually press, not a quarter of the map wrong.

Other than this, I've had a great time working with the library. Thanks for taking the time.

**Comment by johnhult** [comment]
I can also add that if I do
`let a = map.getCenter();
new this.maps.Marker({ position: a, map: map });`

in the init function on googleApiLoaded it does position it correctly. So something seems to be off with the onClick lat and lng.

**Comment by johnhult** [comment]
One more thing is that if I on init instead add a custom click listener
`this.maps.event.addListener(map, 'click', event => {
			this.addNewPoint(event.latLng);
		});`

it works fine.

**Comment by gaudryc** [comment]
Same thing here using all 1.0.x versions and onClick event handler.
There is a systematic shift to the southeast that seems to vary with the zoom level.
The mouse position is good but the associated coordinates are wrong.
Map size is good, Center coordinates is good. Programmatically changing the center coordinates works.
I think something goes wrong in the conversion made by the GeoService object since the refactoring from the version 0.34.
Everything is good using 0.34.

I tried the map listener to illustrate the problem :

  handleMapClick = ({x, y, lat, lng, event}, force) => {
    console.log('==========> handleMapClick(..) : ' + lat + ',' + lng + ' (x=' + x + ',y=' + y + ')');
  }
  handleMapClick2 = ({x, y, lat, lng}) => {
    console.log('==========> handleMapClick2(..) : ' + lat + ',' + lng + ' (x=' + x + ',y=' + y + ')');
  }
  handleMapLoaded = ({ map, maps }) => {
    map.addListener('click', event => {
      this.handleMapClick2({'x': event.pixel.x, 'y': event.pixel.y, 'lat': event.latLng.lat(), 'lng': event.latLng.lng()}); 
    });
  }
...
          <GoogleMap
            ref={ (ref) => { this.map = ref } }
            zoom={ settings.zoom.get() }
            center={ [ latitude, longitude ] }
            onClick={ this.handleMapClick }
            options={ () => this.mapOptions }
            onGoogleApiLoaded={ this.handleMapLoaded }
            yesIWantToUseGoogleMapApiInternals={ true }
            bootstrapURLKeys={{
                key: settings.googleAPIKey.get(),
                language: 'en',
                region: 'en'
            }}>

And I got this :

     ..."==========> handleMapClick2(..) : 48.859025812259674, 2.3398733139038086 (x=749,y=242)"
     ..."==========> handleMapClick(..)  : 48.85902586900235,2.3398694431305103 (x=749,y=242)"

     ..."==========> handleMapClick2(..) : 48.894518236158234, 2.3327064514160156 (x=605,y=37)"
     ..."==========> handleMapClick(..)  : 48.8945535610718,2.3326918518066577 (x=605,y=37)"

Thanks to johnult for this workaround.

**Comment by gaudryc** [comment]
All versions from 0.34 to 1.0.4 are OK.
This issue only applies to version 1.0.5.

**Comment by itsmichaeldiego** [comment]
This was introduced in #552 

@stephenfarrar any idea how we can fix this?

**Comment by melvinsalas** [comment]
I'm going crazy with this issue :(

**Comment by AliMamed** [comment]
onChildMouseMove is allso affected by this problem

**Comment by s123121** [comment]
This bug is driving me crazy for the past 2 days. Thank gods, I finally check out this thread. Apart from downgrading, any idea how to fix this?

**Comment by itsmichaeldiego** [comment]
@stephenfarrar Can you take a look when you have the chance?

**Comment by stephenfarrar** [comment]
Sorry I've been flat out.
I will try to take a look later this week...

**Comment by itsmichaeldiego** [comment]
I just published version `1.0.6` that includes @stephenfarrar's fix: #620 

Thanks @stephenfarrar <3

**Comment by johnhult** [comment]
Works like a charm. Just refactored my code and this fix helped 👍 Feel free to close it.

**Comment by itsmichaeldiego** [comment]
@johnhult Awesome! Sorry for the inconvenient, we try to fix this errors as fast as we can. 

And special thanks to @stephenfarrar ❤️
